x-pack/agent/pkg/agent/application/global_config.go

```diff
@@ -0,0 +1,66 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package application
+
+import (
+	"os"
+	"path/filepath"
+
+	"github.com/elastic/beats/x-pack/agent/pkg/agent/errors"
+	"github.com/elastic/beats/x-pack/agent/pkg/config"
+)
+
+var (
+	homePath string
+	dataPath string
+)
+
+func init() {
+	homePath = retrieveHomePath()
+	dataPath = retrieveDataPath()
+}
+
+// InjectAgentConfig injects config to a provided configuration.
+func InjectAgentConfig(c *config.Config) error {
+	globalConfig := AgentGlobalConfig()
+	if err := c.Merge(globalConfig); err != nil {
+		return errors.New("failed to inject agent global config", err, errors.TypeConfig)
+	}
+
+	return nil
+}
+
+// AgentGlobalConfig gets global config used for resolution of variables inside configuration
+// such as ${path.data}.
+func AgentGlobalConfig() map[string]interface{} {
+	return map[string]interface{}{
+		"path": map[string]interface{}{
+			"data": dataPath,
+			"home": homePath,
+		},
+	}
+}
+
+// retrieveDataPath returns a directory where binary lives
+// Executable is not supported on nacl.
+func retrieveDataPath() string {
+	execPath, err := os.Executable()
+	if err != nil {
+		panic(err)
+	}
+
+	return filepath.Dir(execPath)
+}
+
+// retrieveHomePath returns a home directory of current user
+func retrieveHomePath() string {
+	// this fails usually for not common OSes
+	h, err := os.UserHomeDir()
```

Comment:
@michalpristas This should not be the user's home dir, I think using "." as in libbeat should work here.

Reply:
at the state as it is today I set `path.home` to `$HOME` and `path.data` to where executable lives so basically `.` as we talked on slack.
this is because i wanted to keep one path version/instance independent and shared. so we can specify e.g. download path easier.
In case `path.home` is where exec lives, what do you suggest on using for `path.data`?

Reply:
If we use the default of libbeat, which is indeed a bit different to what you have.

_path.home_ is where the executable is.
_path.data_ is where the beat related data is keep
_path.config_ is where the config is located by default.

See the code here:

https://github.com/elastic/beats/blob/5a12b802870e45de8b8727906f4a6c23070d9f70/libbeat/cfgfile/cfgfile.go#L38-L45

Now this means the following when I take the tarballs, by default everything is *created* in the extracted directory which is the desired behavior.

If you install a package, the service scripts is changing the default behavior and set absolute path, See the debian script by example:

https://github.com/elastic/beats/blob/b6d483d6f150602d43de238b41730d2cd4a6899c/dev-tools/packaging/templates/deb/init.sh.tmpl#L20